python_modules/libraries/dagster-ssh/dagster_ssh/resources.py

```diff
@@ -37,9 +47,49 @@ def key_from_str(key_str):
 
 @beta
 class SSHResource(ConfigurableResource):
-    """Resource for ssh remote execution using Paramiko.
+    """A Dagster resource for establishing SSH connections and performing remote file operations.
+
+    This resource leverages the Paramiko library to provide robust SSH connectivity,
+    including support for key-based and password authentication, tunneling, and SFTP transfers.
+
+    Args:
+        remote_host (str): The hostname or IP address of the remote server to connect to.
+        remote_port (Optional[int]): The SSH port on the remote host. Defaults to standard SSH port 22.
+        username (Optional[str]): The username for SSH authentication. If not provided, defaults to the current system user.
+        password (Optional[str]): The password for SSH authentication. Not recommended for production use; prefer key-based authentication.
+        key_file (Optional[str]): Path to the SSH private key file for authentication.
+        key_string (Optional[str]): SSH private key as a string for authentication.
+        timeout (int, optional): Connection timeout in seconds. Defaults to 10.
+        keepalive_interval (int, optional): Interval for sending SSH keepalive packets. (Defaults to 30 seconds.)
+        compress (bool, optional): Whether to compress the SSH transport stream. Defaults to True.
+        no_host_key_check (bool, optional): Disable host key verification.
+        allow_host_key_change (bool, optional): Allow connections to hosts with changed host keys. (Defaults to False.)
+
+    Example:
+        Creating an SSH resource with key-based authentication:
+
+        .. code-block:: python
+
+            ssh_resource = SSHResource(
+                remote_host='example.com',
```

Comment:
Nit: most examples in our docs use double quotes, so maybe we should do the same here for consistency?

Reply:
thanks updated!